genvalidity-hspec-hashable/test/Test/Validity/HashableSpec.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+-- | Standard 'Spec's for 'Hashable' instances.
+--
+-- You will need @Hashable@ to use these.
+module Test.Validity.HashableSpec where
+
+import Test.Hspec
+import Data.Hashable
+import Test.Validity.Utils
+import GHC.Generics
+
+import Data.GenValidity
+import Test.Validity.Hashable
+
+spec :: Spec
+spec = do
+    hashableSpecOnValid @Double
+    hashableSpec @Int
+    hashableSpecOnArbitrary @Int
+    hashableSpec @HashTrue
+    failsBecause ("Two equal elements aren't hashed to the same value!") $
+        hashableSpec @HashFalse
+
+newtype HashTrue = HashTrue Int
+    deriving (Show, Generic)
+
+hT :: Int -- Number used in the definition of HashTrue
+hT = 7
+
+instance Eq HashTrue where
+    (==) (HashTrue x) (HashTrue y) = (x `mod` hT) == (y `mod` hT)
+
+instance Hashable HashTrue where
+    hashWithSalt n (HashTrue a) = (int ^ expo) `mod` hT
+                               where int = 1 + (a `mod` hT)
```

Comment:
Don't define your own hash function, that's not what you want to test here.
Hashable instances can be defined like `Validity` instances.